rust/otap-dataflow/crates/pdata/src/views/otap/logs.rs

```diff
@@ -0,0 +1,2050 @@
+        match self {
+            Self::Empty => ValueType::Empty,
+            Self::Str(_) => ValueType::String,
+            Self::Int(_) => ValueType::Int64,
+            Self::Double(_) => ValueType::Double,
+            Self::Bool(_) => ValueType::Bool,
+            Self::Bytes(_) => ValueType::Bytes,
+        }
+    }
+
+    fn as_string(&self) -> Option<Str<'a>> {
+        match self {
+            Self::Str(s) => Some(*s),
+            _ => None,
+        }
+    }
+
+    fn as_bool(&self) -> Option<bool> {
+        match self {
+            Self::Bool(b) => Some(*b),
+            _ => None,
+        }
+    }
+
+    fn as_int64(&self) -> Option<i64> {
+        match self {
+            Self::Int(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    fn as_double(&self) -> Option<f64> {
+        match self {
+            Self::Double(d) => Some(*d),
+            _ => None,
+        }
+    }
+
+    fn as_bytes(&self) -> Option<&'a [u8]> {
+        match self {
+            Self::Bytes(b) => Some(*b),
+            _ => None,
+        }
+    }
+
+    fn as_array(&self) -> Option<Self::ArrayIter<'_>> {
+        // TODO: Implement array iteration from CBOR-serialized data
+        None
+    }
+
+    fn as_kvlist(&self) -> Option<Self::KeyValueIter<'_>> {
+        // TODO: Implement map iteration from CBOR-serialized data
+        None
+    }
+}
+
+// ===== Attribute Iterators =====
+
+/// Iterator over log record attributes
+pub struct OtapAttributeIter<'a> {
+    log_attrs: Option<&'a OtapAttributeColumns<'a>>,
+    matching_rows: &'a [usize],
+    current_idx: usize,
+}
+
+impl<'a> OtapAttributeIter<'a> {
+    #[inline]
+    fn new(view: &'a OtapLogsView<'a>, log_row_idx: usize) -> Self {
+        // Get log ID from logs batch
+        let log_id = get_log_id(view.columns.id, log_row_idx);
+
+        let matching_rows = if let Some(id) = log_id {
+            view.log_attrs_map
+                .get(&id)
+                .map(|v| v.as_slice())
+                .unwrap_or(&[])
+        } else {
+            &[]
+        };
+
+        Self {
+            log_attrs: view.log_attrs.as_ref(),
+            matching_rows,
+            current_idx: 0,
+        }
+    }
+}
+
+impl<'a> Iterator for OtapAttributeIter<'a> {
+    type Item = OtapAttributeView<'a>;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current_idx >= self.matching_rows.len() {
+            return None;
+        }
+
+        let attr_row_idx = self.matching_rows[self.current_idx];
+        self.current_idx += 1;
+
+        let log_attrs_cols = self.log_attrs?;
+
+        // Extract key
+        let key = get_attribute_key(log_attrs_cols, attr_row_idx)?;
+
+        // Extract value with type information
+        let value = get_attribute_value(log_attrs_cols, attr_row_idx);
+
+        Some(OtapAttributeView { key, value })
+    }
+}
+
+/// Iterator over attributes for a resource
+pub struct OtapResourceAttributeIter<'a> {
```

Comment:
It seems like `OtapResourceAttributeIter`, `OtapScopeAttributeIter` and `OtapAttributeIter` all have a similar implementation. Is there any reason we couldn't combine these into one type?